js/battle.js

```diff
@@ -1802,6 +1804,41 @@ var Side = (function () {
 		pokemon.statbarElem = this.battle.statElem.children().last();
 		this.updateStatbar(pokemon, true);
 		pokemon.side.updateSidebar();
+
+		if (pokemon.replacedIllusion && !this.battle.hackedPokemon) {
+			if (this.pokemon.indexOf(pokemon) < 0) {
+				// if the pokemon that is switching in isn't in the team, match it to one that is
```

Comment:
I don't think this is the best way to do this, but I don't know how much other things I affect if I would edit `newPokemon` more, and I don't think it's relevant anywhere else.